File: Lista_1/Python/Python/5.py
def functionComplexity(n):
    for i in range(n):
        print("esta é a fn1")
        for j in range(n):
            print("esta é a fn2")
            for k in range(n):
                print("esta é a fn3")

#Supondo que fn1 tem complexidade de tempo O(1) e irá executar n vezes, dizemos que a complexidade
#desta parte do loop é O(n).

#Agora, supondo que a complexidade tempo de fn2 é O(n) e esta irá executar n vezes, dizemos que a
#complexidade desta parte do loop é O(n^2).

#Por fim, como fn3 tem O(n^2) e também irá executar n vezes, a complexidade desta parte do código
#é O(n^3), e por ser a maior e a que predomina, também é a resposta da questão.

#Resposta: A complexidade do algoritmo é O(n^3).

File: Lista_1/C/5.c
#include <stdio.h>

int main()
{
    for(int i = 0; i < n; i++){
        printf("Função fn1()");
        for(int j = 0; j < n; j++){
            printf("Função fn2()");
            for(int k = 0; k < n; k++){
                printf("Função fn3()");
            }
        }
    }
    
/*
Ao saber que a função fn1() possui complexidade O(1) e vai executar n vezes, a complexidade daquela parte do código se torna O(n)
Já a função fn2(), que possui complexiddade O(n), também vai executar n vezes, tornando a complexidade daquela parte O(n^2)
Por fim, a função fn3(), que possui O(n^2), também executando n vezes, vai levar aquela parte do código à possuir O(n^3) e
por ser a complexidade mais relevante, esta é a resposta do exercício. A complexidade de tempo do algoritmo é O(n^3).
*/

    return 0;
}
